pebble/pebble.go

```diff
@@ -0,0 +1,119 @@
+package pebble
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/gofiber/storage/pebble/internal"
+)
+
+type Storage struct {
+	db           *pebble.DB
+	writeOptions *pebble.WriteOptions
+}
+
+type CacheType struct {
+	Data    []byte `json:"data"`
+	Created int64  `json:"created"`
+	Expires int64  `json:"expires"`
+}
+
+func New(config ...Config) *Storage {
+	cfg := configDefault(config...)
+
+	if !internal.IsValid(cfg.Path) {
+		panic(errors.New("invalid filepath"))
+	}
+
+	db, err := pebble.Open(cfg.Path, &pebble.Options{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &Storage{
+		db:           db,
+		writeOptions: cfg.WriteOptions,
+	}
+}
+
+// // Implement the logic to retrieve the value for the given key from the storage provider
+// // Return nil, nil if the key does not exist
+func (s *Storage) Get(key string) ([]byte, error) {
+	if len(key) <= 0 {
+		return nil, nil
+	}
+	data, closer, err := s.db.Get([]byte(key))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, nil
+	}
+
+	if err := closer.Close(); err != nil {
+		log.Fatal(err)
```

Comment:
Not a fan of this, but other storage drivers are doing the same. They should probably be fix in a separate PR

Reply:
In this case I did follow the other storage drivers. It could be changed accordingly.